File: JS/delete_alt.js
//Just trying to say whether javascript can be used as a 
//general purpose programming language
function Node(value){
	this.value = value;
	this.next = null;
};

function SinglyLinkedList(){
	this.contruct = function(value){
		this.head = new Node(value);
		this.tail = this.head;
	};

	this.addnode = function(value){
		this.tail.next = new Node(value);
		this.tail = this.tail.next;
	};

	this.print = function(){
		this.cur = this.head;
		while(this.cur != null){
			print(this.cur.value);
			this.cur = this.cur.next;
		};
	}; 

	this.deletealt = function(){
		this.cur = this.head;
		while(this.cur.next != null){
			this.cur.next = this.cur.next.next;
			print(this.cur.value);
			this.cur = this.cur.next;
		}
	} 
};

var newObject = new SinglyLinkedList();
newObject.contruct(1);
newObject.addnode(2);
newObject.addnode(3);
newObject.addnode(4);
newObject.addnode(5);
newObject.deletealt();

File: scala/delete_alt.java
class Node{
	int data;
	Node next;

	public Node(int data){
		data = data;
		next = null;
	}
}

class SinglyLinkedList{
	Node head;
	Node tail;

	public SinglyLinkedList(){
		head = null;
		tail = null;
	}

	public void addNode(int data){
		if(head == null){
			head = new Node(data);
			tail = head;
		}
		else
		{
			tail.next = new Node(data);
		}
	}

	public void printList(){
		Node cur = head;
		while(cur != null){
			System.out.println(cur.data);
			cur = cur.next;
		}
	}


	public static void main(String[] args){
		SinglyLinkedList single = new SinglyLinkedList();
		single.addNode(1);
		single.addNode(2);
		single.addNode(3);
		single.addNode(4);
		single.addNode(5);
		single.printList();
	}
}